feat(getting-started): Add getting started guide

tymokvo opened the pull request.

tymokvo commented:
Ok, @mostaphaRoudsari I moved the dev guide here. I think it is mostly done, but could use some feedback. I'm also waiting to figure out why the recipe is not working with multiple models right now. Once that is resolved, I will move it out of draft status.

I just read through the comments on the original proposal again and it seems that maybe this should not include a section on using `queenbee-pollination`? I may have misunderstood the conversation there. If that's the case I will remove those sections.

It is probably easiest to see it at [this link](https://github.com/tymokvo/guides/tree/add-dev-guide/getting-started) as that is how users should be seeing it eventually.

mostaphaRoudsari commented:
Thank you! I'll check this out in the evening.

tymokvo commented:
Ok, I fixed the job parameters, so it should be possible to just follow the directions all the way through now.

mostaphaRoudsari commented on getting-started/README.md:
```diff
@@ -0,0 +1,408 @@
+
+#### Create Artifacts
+
+The two model files that were downloaded above need to be submitted as
```
I think adding some opening statements like this can help for people to connect the steps together:

Now that we have the project set-up and the recipe we need to add the artifacts to project in order to use them as input arguments for the recipe

tymokvo replied:
Yes agreed

tymokvo replied:
done

mostaphaRoudsari commented on getting-started/README.md:
```diff
@@ -0,0 +1,408 @@
+### Fin
+
+And that's it! You have now created a project and used it to create a
+parametric job to calculate the annual daylight factor of two different 3D
```
```suggestion
parametric job to calculate the daylight factor of two different 3D
```

tymokvo replied:
I just combine all of the recipe names "annual daylight factor sun hours energy use comfort map..." :sunglasses:

mostaphaRoudsari commented on getting-started/README.md:
```diff
@@ -0,0 +1,408 @@
+parametric job to calculate the annual daylight factor of two different 3D
+models simultaneously!
+
+## Use a Client SDK
```
Should we add 'To be added' or some note like that here?

tymokvo replied:
Sure, if that is ok. I am not sure what the status of SDKs is but from the comments on Basecamp it sounded like maybe its not ready right now?

mostaphaRoudsari replied:
The one for C# is working for sure. We haven't tested the one for Python so it is probably safe to leave it out for now since your code is easy to understand and basically does the same.

tymokvo replied:
done

mostaphaRoudsari reviewed (commented):
I read this document with a smile. It's really good. I left some minor comments.

Didn't test the code on my side - maybe I do that tomorrow.

mostaphaRoudsari commented on getting-started/rest-api/requirements.txt:
```diff
@@ -0,0 +1,34 @@
+Pygments==2.8.1
+pyrsistent==0.17.3
+PyYAML==5.4.1
+queenbee==1.24.10
```
I forgot about this one last night! Do we really need all these dependencies for running this code? I'm concerned that using all the `==` will cause issues for users who are testing it. I know that you recommended running it in a virtual env but from my experience not everyone knows who to do that!

1. We should probably provide a link to show them how to do that?
2. Minimize the number dependencies if possible.
3. Use >= instead of == whenever possible.

tymokvo replied:
We may not need all of them, I can try to reduce it. I probably have some unnecessary dev dependencies in there.

But, maybe we should make a Teachable video of the process to supplement? I would hate to have someone get stuck because they have some dependency conflict that arises from future library updates and virtualenvs are supposed to help alleviate that.

mostaphaRoudsari replied:
A video is actually a great idea to go with the text. We can add it as a free video on YouTube. I just don't want it to be a distraction for you.

tymokvo replied:
Sure, it's maybe not important right now. Perhaps in a couple months when we are getting more people on the platform.

tymokvo replied:
done

tymokvo commented on getting-started/README.md:
```diff
@@ -0,0 +1,408 @@
+
+## Prerequisites
+
+1. Have an active [app.pollination.cloud][poll-app] account
+1. Some familiarity with command line tools and scripting
+1. Familiarity with Python 3 will help, but is not required
+
+[poll-app]: https://app.pollination.cloud
```
I prefer the footnote style for anything longer than a couple words. It also makes it reusable in other places in the document.

tymokvo commented on getting-started/README.md:
```diff
@@ -0,0 +1,408 @@
+
+Remember:
+
+**API Keys are like passwords!**
```
Lol, good point. More emojis -> better document. And thanks!

tymokvo replied:
done

tymokvo commented on getting-started/README.md:
```diff
@@ -0,0 +1,408 @@
+
+We create a public project with a meaningful name and description. The name
+will be necessary to create further resources tied to the project, so this is a
+useful object to keep around.
```
Haha, I didn't think of that. Will change.

tymokvo replied:
done

tymokvo commented on getting-started/README.md:
````diff
@@ -0,0 +1,408 @@
+useful object to keep around.
+
+```python
+project = Payload.Create(
````
Yes, I assumed the reader had my context in their head :brain: . Will update.

tymokvo replied:
done

tymokvo commented on getting-started/README.md:
```diff
@@ -0,0 +1,408 @@
+`custom-application`, but you may want to use something more meaningful for
+your purposes), and click `Add` to generate a key for this name.
+
+![Settings](./assets/settings.png)
```
Yeah I was going to edit the images as the last step to crop them down, add annotations, etc. But I didn't want to spend time editing until the text is finished.

tymokvo commented on getting-started/README.md:
```diff
@@ -0,0 +1,408 @@
+
+This guide has three sections:
+
+1. Obtain an API key
```
done

tymokvo commented on getting-started/README.md:
```diff
@@ -0,0 +1,408 @@
+We will use two models two show how to run the same simulation with two options
+for a facade system. The models are included as siblings to the Python source
+files in the [`rest-api`](./rest-api) folder.
+
```
done

tymokvo commented on getting-started/README.md:
```diff
@@ -0,0 +1,408 @@
+We will use the library `httpx` to create a client that inserts this header for
+all requests. You will need to `pip install httpx`.
+
+The source for this script is can be found in [`get_user.py`](./get_user.py).
```
done

tymokvo commented on getting-started/README.md:
```diff
@@ -0,0 +1,408 @@
+Once you have signed in, you will see the Pollination dashboard. In the top
+right, click on your avatar to reveal the `Profile` link.
+
+![Dashboard](./assets/dashboard.png)
```
True! I will update.

tymokvo commented:
I think I've addressed all comments! Feel free to unresolve anything though. If the text/code seems good, I will edit the images to be more presentable.

mostaphaRoudsari reviewed (approved):
LGTM!

Let's merge this in after updating the images and I will include it as a resource for early access. Thanks @tymokvo!

PS: If you think updating the images will take time them let's merge this in and we can update them in a separate PR.

tymokvo commented:
Ok! I created #2  to follow up on the images as I think I should focus on the backend refactor today.
